src/shared/components/LomFieldsInput/LomFieldsInput.tsx

```diff
@@ -0,0 +1,170 @@
+import { Form, FormGroup, Spacer, TagsInput } from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import { differenceWith, isEmpty, isEqual, isNil, map, without } from 'lodash-es';
+import React, { FC, useEffect, useState } from 'react';
+
+import { CustomError } from '../../helpers';
+import useTranslation from '../../hooks/useTranslation';
+import { LomService } from '../../services/lom.service';
+import { LomFieldsByScheme } from '../../types/lom';
+
+import { mapLomFieldsToOptions, mapOptionsToLomFields } from './LomFieldsInput.helpers';
+
+type LomFieldsInputProps = {
+	contexts: Avo.Lom.LomField[];
+	educationLevels: Avo.Lom.LomField[];
+	subjects: Avo.Lom.LomField[];
+	themes: Avo.Lom.LomField[];
+	onChange: (lomFields: Partial<LomFieldsByScheme>) => void;
+};
+
+const LomFieldsInput: FC<LomFieldsInputProps> = ({
+	contexts,
+	educationLevels,
+	subjects,
+	themes,
+	onChange,
+}) => {
+	const { tText } = useTranslation();
+	const [educationLevelsOptions, setEducationLevelsOptions] = useState<Avo.Lom.LomField[]>([]);
+	const [subjectsOptions, setSubjectsOptions] = useState<Avo.Lom.LomField[]>([]);
+	const [themesOptions, setThemesOptions] = useState<Avo.Lom.LomField[]>([]);
+
+	useEffect(() => {
+		let fetch = true;
+
+		LomService.fetchEducationLevels()
+			.then((res: Avo.Lom.LomField[]) => {
+				fetch && setEducationLevelsOptions(res);
+			})
+			.catch((err) => {
+				console.error(
+					new CustomError('Failed to get education levels from the database', err)
+				);
+			});
+
+		return () => {
+			fetch = false;
+		};
+	}, [setEducationLevelsOptions]);
+
+	useEffect(() => {
+		let fetch = true;
+
+		LomService.fetchSubjects(map(subjects, 'id'))
+			.then((res: Avo.Lom.LomField[]) => {
+				fetch && setSubjectsOptions(res);
+			})
+			.catch((err) => {
+				console.error(new CustomError('Failed to get subjects from the database', err));
+			});
+
+		return () => {
+			fetch = false;
+		};
+	}, [setSubjectsOptions]);
+
+	useEffect(() => {
+		let fetch = true;
+
+		LomService.fetchThemes(map(themes, 'id'))
+			.then((res: Avo.Lom.LomField[]) => {
+				fetch && setThemesOptions(res);
+			})
+			.catch((err) => {
+				console.error(new CustomError('Failed to get themes from the database', err));
+			});
+
+		return () => {
+			fetch = false;
+		};
+	}, [setThemesOptions]);
```

Comment:
wouldn't it be easier to wrap these service functions in a react-query useQuery wrapper and then you get the isLoading and isError booleans for free?